homeassistant/components/advantage_air/climate.py

```diff
@@ -84,34 +103,56 @@ class AdvantageAirAC(AdvantageAirAcEntity, ClimateEntity):
     _attr_min_temp = 16
     _attr_name = None
     _enable_turn_on_off_backwards_compatibility = False
+    _support_preset = ClimateEntityFeature(0)
 
     def __init__(self, instance: AdvantageAirData, ac_key: str) -> None:
         """Initialize an AdvantageAir AC unit."""
         super().__init__(instance, ac_key)
 
-        self._attr_supported_features = (
-            ClimateEntityFeature.FAN_MODE
-            | ClimateEntityFeature.TURN_OFF
-            | ClimateEntityFeature.TURN_ON
-        )
-        self._attr_hvac_modes = [
-            HVACMode.OFF,
-            HVACMode.COOL,
-            HVACMode.HEAT,
-            HVACMode.FAN_ONLY,
-            HVACMode.DRY,
-        ]
-        # Set supported features and HVAC modes based on current operating mode
+        self._attr_preset_modes = [ADVANTAGE_AIR_MYZONE]
+
+        # Add "MyTemp" preset if available
+        if ADVANTAGE_AIR_MYTEMP_ENABLED in self._ac:
+            self._attr_preset_modes += [ADVANTAGE_AIR_MYTEMP]
+            self._support_preset = ClimateEntityFeature.PRESET_MODE
+
+        # Add "MyAuto" preset if available
+        if ADVANTAGE_AIR_MYAUTO_ENABLED in self._ac:
+            self._attr_preset_modes += [ADVANTAGE_AIR_MYAUTO]
+            self._support_preset = ClimateEntityFeature.PRESET_MODE
+
+        # Setup attributes based on current preset
+        self.configure_preset()
+
+    def configure_preset(self) -> None:
```

Comment:
```suggestion
        self._async_configure_preset()

    def _async_configure_preset(self) -> None:
```

Because we have both sync and async code in this codebase, anything that is safe to run in the event loop is usually prefixed with `async_` or `_async_`